ui/App.jsx

```diff
@@ -55,15 +56,24 @@ const updateOldUrls = () => {
     if (index >= 2 && updates.pathname !== subRoute && subRoute !== '/jobs') {
       updates.pathname += subRoute;
     }
-  } else if (search.length) {
+  }
+
+  if (search.length) {
```

Comment:
I don't think this change is intended. We should keep this behavior under the `else`, as before